TypeCobol.Test/Utils/Comparator.cs

```diff
@@ -1180,33 +1180,65 @@ public ChangeComparator(Paths path, bool debug = false, bool isEI = false)
 
         }
 
-        public override void Compare(CompilationUnit result, StreamReader reader, string expectedResultPath)
+        public override void Compare(CompilationUnit result, IncrementalChangesHistory history, StreamReader reader, string expectedResultPath)
         {
-            // Dump latest incremental changes for the given snapshots
+            // Dump full incremental history
             var actual = new StringBuilder();
-            DumpChanges(result.CobolTextLinesVersion, DumpCobolTextLine);
-            DumpChanges(result.TokensDocumentSnapshot.CurrentVersion, DumpTokensLine);
-            DumpChanges(result.ProcessedTokensDocumentSnapshot.CurrentVersion, DumpProcessedTokensLine);
-            DumpChanges(result.CodeElementsDocumentSnapshot.CurrentVersion, DumpCodeElementsLine);
+            DumpEvents(history.TextChangedEvents, DumpCobolTextLine);
+            DumpEvents(history.TokensChangedEvents, DumpTokensLine);
+            DumpEvents(history.ProcessedTokensChangedEvents, DumpProcessedTokensLine);
+            DumpEvents(history.CodeElementsChangedEvents, DumpCodeElementsLine);
 
             // Compare with expected
             ParserUtils.CheckWithResultReader(paths.SamplePath, actual.ToString(), reader, expectedResultPath);
 
-            void DumpChanges<TLine>(DocumentVersion<TLine> version, Action<StringBuilder, TLine> dumpLine)
+            void WriteSeparator(string title, char separatorChar = '=')
+            {
+                int paddingLength = 78 - title.Length;
+                string left = new string(separatorChar, paddingLength / 2);
+                string right = new string(separatorChar, paddingLength / 2 + paddingLength % 2);
+                actual.AppendLine($"{left} {title} {right}");
+            }
+
+            void DumpEvents<TLine>(IEnumerable<DocumentChangedEvent<TLine>> changeEvents, Action<StringBuilder, TLine> dumpLine)
```

Comment:
As discussed, can you dump only the last and last-1 elements of `changeEvents` ?